chainer/distributions/bijector.py

```diff
@@ -0,0 +1,88 @@
+from chainer.functions.math import exponential
+from chainer.functions.math import identity
+
+
+class Bijector():
+
+    """Interface of Bijector.
+
+    `Bijector` is implementation of bijective (invertible) function that is
+    used by `TransformedDistribution`. The three method `_forward`, `_inv` and
+    `_logdet_jac` have to be defined in inhereted class.
+    """
+
+    def __init__(self):
+        self.cache_x_y = (None, None)
+
+    def forward(self, x):
+        old_x, old_y = self.cache_x_y
+        if id(x) == id(old_x):
+            return old_y
+        y = self._forward(x)
+        self.cache_x_y = x, y
+        return y
+
+    def inv(self, y):
+        old_x, old_y = self.cache_x_y
+        if id(y) == id(old_y):
+            return old_x
+        x = self._inv(y)
```

Comment:
`y` can be a `ndarray` with the current implementation of `TransformedDistribution.cdf`, while `self._inv` requires `Variable`.

Reply:
I fix docstring as like that `self._inv` requires `Variable` and `ndarray`.